apps/apollo/src/connectors/users.test.ts

```diff
@@ -1,65 +1,89 @@
-/* eslint-disable @typescript-eslint/no-unsafe-call -- test conveniency */
-/* eslint-disable @typescript-eslint/no-unsafe-return -- test conveniency */
-/**
- * DISCLAIMER:
- * The tests provided in this file are specifically designed for the `auth` connectors function.
- * Theses tests exists because the services & inngest functions using this connector mock it.
- * If you are using an SDK we suggest you to mock it not to implements calls using msw.
- * These file illustrate potential scenarios and methodologies relevant for SaaS integration.
- */
-
-import { http } from 'msw';
 import { describe, expect, test, beforeEach } from 'vitest';
+import { http } from 'msw';
 import { server } from '../../vitest/setup-msw-handlers';
-import { type MySaasUser, getUsers } from './users';
-import { MySaasError } from './commons/error';
+import { getUsers, deleteUser,type ApolloUser, type Pagination } from './users';
+import  { ApolloError } from './commons/error';
 
-const validToken = 'token-1234';
-const maxPage = 3;
+const users: ApolloUser[] = [
+  {
+    id: 'user-id',
+    name: 'Username',
+    email: 'user@gmail.com',
+  },
+];
 
-const users: MySaasUser[] = Array.from({ length: 5 }, (_, i) => ({
-  id: `id-${i}`,
-  username: `username-${i}`,
-  email: `user-${i}@foo.bar`,
-}));
+const pagination: Pagination = {
+  page: "1",
+  per_page: 10,
+  total_entries: 2,
+  total_pages: 1,
+};
 
-describe('auth connector', () => {
-  describe('getUsers', () => {
-    // mock token API endpoint using msw
-    beforeEach(() => {
-      server.use(
-        http.get('https://mysaas.com/api/v1/users', ({ request }) => {
-          // briefly implement API endpoint behaviour
-          if (request.headers.get('Authorization') !== `Bearer ${validToken}`) {
-            return new Response(undefined, { status: 401 });
-          }
-          const url = new URL(request.url);
-          const pageParam = url.searchParams.get('page');
-          const page = pageParam ? Number(pageParam) : 0;
-          if (page === maxPage) {
-            return Response.json({ nextPage: null, users });
-          }
-          return Response.json({ nextPage: page + 1, users });
-        })
-      );
-    });
+const validToken = 'test-token';
+const userId= "test-id";
+const maxPage = "2"; 
+describe('getUsers', () => {
+  beforeEach(() => {
+    server.use(
+      http.get('https://api.apollo.io/v1/users/search', ({ request }) => {
+        const url = new URL(request.url);
+        if (url.searchParams.get('api_key') !== validToken) {
+          return new Response(undefined, { status: 401 });
+        }
+        const page = parseInt(url.searchParams.get('page') || "0");
+        const response = {
```

Comment:
it's better if we directly return the response instead of creating a new variable.